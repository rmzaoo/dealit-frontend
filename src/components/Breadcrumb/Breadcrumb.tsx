import React, { useState, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import { useMainCategoryBySubFetcher } from "../../hooks/categories/useMainCategoryBySubFetcher";
import {
  BreadCrumbAnchor,
  BreadCrumbContainer,
  BreadCrumbSpan,
  BreadCrumbText,
} from "./style";
import { MdKeyboardArrowRight } from "react-icons/md";

interface Props {
  subCategory: string;
}

const Breadcrumb = ({ subCategory }: Props) => {
  const mainCat = useMainCategoryBySubFetcher(subCategory)?.main.name;
  const navigate = useNavigate();
  console.log(useMainCategoryBySubFetcher(subCategory)?.main.name);

  return (
    <BreadCrumbContainer>
      <BreadCrumbText>
        <BreadCrumbAnchor onClick={() => navigate(`../products/${mainCat}`)}>
          {mainCat}
        </BreadCrumbAnchor>
      </BreadCrumbText>
      <BreadCrumbSpan>
        <MdKeyboardArrowRight
          style={{ color: "gray", verticalAlign: "middle" }}
        />
      </BreadCrumbSpan>
      <BreadCrumbText>
        <BreadCrumbAnchor
          onClick={() => navigate(`../products/${mainCat}/${subCategory}`)}
        >
          {subCategory}
        </BreadCrumbAnchor>
      </BreadCrumbText>
    </BreadCrumbContainer>
  );
};

export default Breadcrumb;
