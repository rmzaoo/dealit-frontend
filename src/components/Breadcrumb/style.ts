import styled from "styled-components";

export const BreadCrumbContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
`;
export const BreadCrumbText = styled.p`
  font-size: 0.9rem;
  padding-left: 2px;
  color: gray;

  //Tablets
  @media screen and (max-width: 768px) {
    font-size: 0.8rem;
  }

  //Mobile
  @media screen and (max-width: 480px) {
    font-size: 0.7rem;
  }
`;

export const BreadCrumbAnchor = styled.a`
  text-decoration: none;
  cursor: pointer;

  &:hover {
    color: #02a4e3;
  }
`;

export const BreadCrumbSpan = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
`;
