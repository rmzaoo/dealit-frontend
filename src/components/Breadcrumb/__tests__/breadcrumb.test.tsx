import React from "react";
import { describe, test, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Breadcrumb from "../Breadcrumb";
import { BrowserRouter } from "react-router-dom";

describe("Breadcrumb", () => {
  test("Renders whithout crashing.", () => {
    const result = render(
      <BrowserRouter>
        <Breadcrumb subCategory={"subCategory1"} />
      </BrowserRouter>
    );
    expect(result).toMatchSnapshot();
  });
});
