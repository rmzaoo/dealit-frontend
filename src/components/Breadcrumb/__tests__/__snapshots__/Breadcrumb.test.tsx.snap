// Vitest Snapshot v1

exports[`Breadcrumb > Renders whithout crashing. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ cuOToI"
      >
        <p
          class="sc-gsnTZi kuxFnl"
        >
          <a
            class="sc-dkzDqf cSVege"
          />
        </p>
        <span
          class="sc-hKMtZM dYvYDs"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: gray; vertical-align: middle;"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </span>
        <p
          class="sc-gsnTZi kuxFnl"
        >
          <a
            class="sc-dkzDqf cSVege"
          >
            subCategory1
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ cuOToI"
    >
      <p
        class="sc-gsnTZi kuxFnl"
      >
        <a
          class="sc-dkzDqf cSVege"
        />
      </p>
      <span
        class="sc-hKMtZM dYvYDs"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: gray; vertical-align: middle;"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </span>
      <p
        class="sc-gsnTZi kuxFnl"
      >
        <a
          class="sc-dkzDqf cSVege"
        >
          subCategory1
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
